lib/Sema/CSDiag.cpp

```diff
@@ -7881,12 +7881,13 @@ static void noteArchetypeSource(const TypeLoc &loc, ArchetypeType *archetype,
     // `Pair<Any, Any>`.
     // Right now we only handle this when the type that's at fault is the
     // top-level type passed to this function.
-    if (loc.getType().isNull()) {
-      return;
-    }
-    
+    auto type = loc.getType();
+    if (!type)
+      type = cs.getType(loc);
```

Comment:
Why is this directly calling `loc.getType()` and then only if getting `null` calling `cs.getType(loc)`?